transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/backends/BackendRegistry.java

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.backends;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.content.pm.ServiceInfo;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/** Container for registered {@link TransportBackend}s. */
+@Singleton
+public class BackendRegistry {
+
+  private static final String TAG = "BackendRegistry";
+  private static final String BACKEND_KEY_PREFIX = "backend:";
+
+  private final Context applicationContext;
+  private final CreationContextFactory creationContext;
+  private final Map<String, TransportBackend> backends = new HashMap<>();
+  private Map<String, String> backendProviders = null;
+
+  @Inject
+  public BackendRegistry(Context applicationContext, CreationContextFactory creationContext) {
+    this.applicationContext = applicationContext;
+    this.creationContext = creationContext;
+  }
+
+  @Nullable
+  public synchronized TransportBackend get(String name) {
+    if (backends.containsKey(name)) {
+      return backends.get(name);
+    }
+    String backendProviderName = getBackendProviders().get(name);
+    if (backendProviderName == null) {
+      return null;
+    }
+
+    try {
+      BackendFactory provider =
+          (BackendFactory)
```

Comment:
And do TransportBackend provider = (TransportBackend)  Class.forName(backendProviderName).getInstance() ?

Reply:
see above